internal/app.go

```diff
@@ -10,44 +10,56 @@ package internal
 import (
 	"context"
 	"log/slog"
-	"os"
-	"time"
 
 	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/config"
+	"github.com/nginx/agent/v3/internal/logger"
 	"github.com/nginx/agent/v3/internal/plugin"
+	"github.com/spf13/cobra"
 )
 
-type App struct{}
+type App struct {
+	commit  string
+	version string
+}
 
-func NewApp() *App {
-	return &App{}
+func NewApp(commit, version string) *App {
+	return &App{commit, version}
 }
 
-func (*App) Run() error {
-	ctx := context.Background()
+func (a *App) Run() error {
```

Comment:
The one letter method receiver is, unfortunately, idiomatic to Go: https://go.dev/wiki/CodeReviewComments#receiver-names

But I'm okay with breaking convention for this case, since `app` is such a short word anyway.

Reply:
I'd be more inclined to follow the convention in Go just to be consistent.